spyder/plugins/editor/extensions/closebrackets.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+"""This module contains the close quotes editor extension."""
+
+from qtpy.QtCore import Qt
+from qtpy.QtGui import QTextCursor
+
+from spyder.api.editorextension import EditorExtension
+
+
+class CloseBracketsExtension(EditorExtension):
+    """Editor Extension for insert brackets automatically."""
+    BRACKETS_LIST = ["(", ")", "{", "}", "[", "]"]
+    BRACKETS_KEYS = [Qt.Key_ParenLeft, Qt.Key_ParenRight,
+                     Qt.Key_BraceLeft, Qt.Key_BraceRight,
+                     Qt.Key_BracketLeft, Qt.Key_BracketRight]
+    BRACKETS_CHAR = dict(zip(BRACKETS_KEYS, BRACKETS_LIST))
+    BRACKETS_LEFT = dict(zip(BRACKETS_KEYS[::2], BRACKETS_LIST[::2]))
+    BRACKETS_RIGHT = dict(zip(BRACKETS_KEYS[1::2], BRACKETS_LIST[1::2]))
+    BRACKETS_PAIR = {Qt.Key_ParenLeft: "()", Qt.Key_ParenRight: "()",
+                     Qt.Key_BraceLeft: "{}", Qt.Key_BraceRight: "{}",
+                     Qt.Key_BracketLeft: "[]", Qt.Key_BracketRight: "[]"}
+
+    def on_state_changed(self, state):
+        """Connect/disconnect sig_key_pressed signal."""
+        if state:
+            self.editor.sig_key_pressed.connect(self._on_key_pressed)
+        else:
+            self.editor.sig_key_pressed.disconnect(self._on_key_pressed)
+
+    def _on_key_pressed(self, event):
+        if event.isAccepted():
+            return
+
+        key = event.key()
+        if key in self.BRACKETS_CHAR and self.enabled:
+            self._autoinsert_brackets(key)
+            event.accept()
+
+    def unmatched_brackets_in_line(self, text, closing_braces_type=None):
+        """
+        Checks if there is an unmatched brackets in the 'text'.
+        The brackets type can be general or specified by closing_braces_type
+        (')', ']' or '}')
+        """
+        if closing_braces_type is None:
+            opening_braces = self.BRACKETS_LEFT.values()
+            closing_braces = self.BRACKETS_RIGHT.values()
```

Comment:
Should presumably be "brackets", not braces, throughout this block, to be consistent with the verbiage used elsewhere, as well as to be semantically correct (since "braces" specifically refers to ``{}`` while "brackets" is the umbrella term for all types).

Reply:
Yeah I agree. I didn't get a chance to clean that part up yet. There are a lot of different terms in the current code base.